### DISCUSSION TITLE
课程5：AttributeError: module 'tvm.meta_schedule' has no attribute 'TuneConfig'这是什么问题

### DISCUSSION THREAD

**Discussion by JJYGood** [mlc-ai/mlc-zh#162] [Q&A]
from tvm import meta_schedule as ms

sch_tuned = ms.tune_tir(
    mod=MyModule,
    target="llvm --num-cores=1",
    config=ms.TuneConfig(
      max_trials_global=64,
      num_trials_per_iter=64,
    ),
    space=ms.space_generator.ScheduleFn(stochastic_schedule_mm),
    work_dir="./tune_tmp",
    task_name="main"
)

---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
[<ipython-input-30-8d81599fb5f8>](https://localhost:8080/#) in <module>
      4     mod=MyModule,
      5     target="llvm --num-cores=1",
----> 6     config=ms.TuneConfig(
      7       max_trials_global=64,
      8       num_trials_per_iter=64,

AttributeError: module 'tvm.meta_schedule' has no attribute 'TuneConfig'

**Comment by ticoAg** [comment]
应该是使用的tvm版本有差异 当前默认装的版本$max_trials_global$和$num_trials_per_iter$直接作为tune_tir的参数 
run默认的代码需在colab执行


```python
!wget https://github.com/mlc-ai/utils/releases/download/v0.9.dev0/mlc_ai_nightly_cu110-0.9.dev2227%2Bg06c0ef3be-cp37-cp37m-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
!pip install ./mlc_ai_nightly_cu110-0.9.dev2227+g06c0ef3be-cp37-cp37m-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
```

**Comment by JJYGood** [comment]
感谢，之后装了带有cuda版本的是能成功的




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2022年11月7日(星期一) 上午10:42
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [mlc-ai/mlc-zh] 课程5：AttributeError: module &#39;tvm.meta_schedule&#39; has no attribute &#39;TuneConfig&#39;这是什么问题 (Discussion #162)





 
应该是使用的tvm版本有差异 当前默认装的版本$max_trials_global$和$num_trials_per_iter$直接作为tune_tir的参数
 run默认的代码需在colab执行
 !wget https://github.com/mlc-ai/utils/releases/download/v0.9.dev0/mlc_ai_nightly_cu110-0.9.dev2227%2Bg06c0ef3be-cp37-cp37m-manylinux_2_17_x86_64.manylinux2014_x86_64.whl !pip install ./mlc_ai_nightly_cu110-0.9.dev2227+g06c0ef3be-cp37-cp37m-manylinux_2_17_x86_64.manylinux2014_x86_64.whl

**Comment by wangyeahua** [comment]
API变了， 最新API可以参考tvm/python/tvm/meta_schedule/tir_integration.py 的134行tune_tir的定义。
可以修改你的代码如下：
``
db = ms.tune_tir(
    mod=MyModule,
    target="llvm --num-cores=1",
    # config=ms.TuneConfig(
    #   max_trials_global=64,
    #   num_trials_per_iter=64,
    # ),
    max_trials_global=64,
    num_trials_per_iter=64,
    space=ms.space_generator.ScheduleFn(stochastic_schedule_mm),
    work_dir="./tune_tmp",
    task_name="main"
)
tuned_sch=ms.tir_integration.compile_tir(db,MyModule,target="llvm")
